test/singa/test_textfile_decoder.cc

```diff
@@ -0,0 +1,61 @@
+/************************************************************
+*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*************************************************************/
+
+#include "singa/io/encoder.h"
+#include "singa/io/decoder.h"
+#include "gtest/gtest.h"
+
+using singa::Shape;
+using singa::Tensor;
+TEST(TextDecoder, Decode) {
+  singa::TextEncoder encoder;
+  singa::TextDecoder decoder;
+  
+  std::stringstream line("2, 1.34, 4.5, 6., 7.45, 1.0");
+  int raw_label = 0;
+  float* raw_data = new float[5];
+  line >> raw_label;
+  size_t size = 0;
+  std::string str;
+  while (std::getline(line, str, ',')) {
+    float temp;
+    if (std::stringstream(str) >> temp)
+      raw_data[size++] = temp;
+  }
```

Comment:
you can test the encoder in this way,
1. create a tensor for data and a tensor for label
2. encoder them into string.
3. compare the encoded with the string created by you.

The decoder can be test in 3, 2 (code), 1